src/common/tests_dmg_helpers.c

```diff
@@ -415,12 +415,12 @@ dmg_pool_create(const char *dmg_config_file,
 	if (prop != NULL) {
 		entry = daos_prop_entry_get(prop, DAOS_PROP_PO_ACL);
 		if (entry != NULL) {
-			tmp_name = tmpnam(tmp_buf);
-			if (tmp_name == NULL) {
-				D_ERROR("failed to create tmpfile name\n");
+			fd = mkstemp(tmp_name);
+			if (fd < 0) {
+				D_ERROR("failed to create tmpfile file\n");
 				D_GOTO(out_cmd, rc = -DER_NOMEM);
 			}
-			tmp_file = fopen(tmp_name, "w");
+			tmp_file = fdopen(fd, "w");
 			if (tmp_file == NULL) {
 				D_ERROR("failed to open %s\n", tmp_name);
 				D_GOTO(out_cmd, rc = -DER_MISC);
```

Comment:
Are we leaking the open file handle here?   do we need a call to close(fd) when this fails?  It appears that fclose will close the file descriptor.

Reply:
@jolivier23 I think you are right, even if it is very unlikely to happen as the file is already open, I don't know what would cause fdopen() to fail ! But then, I think that the error msg may also need to be changed to reflect that this is the fd to stream conversion which has failed, and printing errno might also be interesting to print in this case...